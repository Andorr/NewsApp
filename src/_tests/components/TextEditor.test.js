import React from 'react';
import {shallow, mount} from 'enzyme';
import { createShallow , createMount, type ReactWrapper } from '@material-ui/core/test-utils';

// Component imports
import Tabs from '@material-ui/core/Tabs';
import TextEditor, {ToolbarAction} from '../../components/inputs/TextEditor';

describe('Testing TextEditor', () => {
    let shallow;
    let mount;

    beforeEach(() => {
        shallow = createShallow(); // Necassary for Material-UI's "withStyles()"
        mount = createMount();
    });

    it('[TextEditor] Testing render', () => {
        const component = shallow(<TextEditor />);
        expect(component).toMatchSnapshot();
    });

    it('[TextEditor] Testing "appendTextAtCursor()"', () => {
        let value = '';
        const onChange = jest.fn((newValue) => value = newValue);
        const component: ReactWrapper = mount(<TextEditor value={value} onChange={onChange}/>);
        component.find(ToolbarAction).first().simulate('click'); // Should call appendTextAtCursor()
        expect(onChange).toBeCalled();
        expect(value).toBe('### ');
    });

    it('[TextEditor] Testing disablePreview prop', () => {
        const componentWithPreview: ReactWrapper = mount(<TextEditor />);
        expect(componentWithPreview.find(Tabs).exists()).toBeTruthy();
        componentWithPreview.unmount();
        const componentWithoutPreview: ReactWrapper = mount(<TextEditor disablePreview />);
        expect(componentWithoutPreview.find(Tabs).exists()).toBeFalsy();
    });
});