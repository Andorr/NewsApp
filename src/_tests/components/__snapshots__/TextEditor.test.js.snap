// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TextEditor [TextEditor] Testing render 1`] = `
<TextEditor
  classes={
    Object {
      "emptySpace": "TextEditor-emptySpace-7",
      "iconBtn": "TextEditor-iconBtn-6",
      "input": "TextEditor-input-2",
      "preview": "TextEditor-preview-4",
      "root": "TextEditor-root-1",
      "textBtn": "TextEditor-textBtn-5",
      "toolbox": "TextEditor-toolbox-3",
    }
  }
/>
`;
